sympy/functions/special/bsplines.py

```diff
@@ -202,3 +202,73 @@ def bspline_basis_set(d, knots, x):
     """
     n_splines = len(knots) - d - 1
     return [bspline_basis(d, knots, i, x) for i in range(n_splines)]
+
+
+def interpolating_spline(d, x, X, Y):
+    """A spline of degree ``d`` interpolating the points with given ``X`` and ``Y`` coordinates.
```

Comment:
Could you please keep the docstring lines less than 72 characters? Note: in the doctests themselves, you can wrap long output -- I believe whitespace will be ignored.

```
>>> foo
Piecewise((3*x, (x >= 1) & (x <= 2)),
    (-x/2 + 7, (x >= 2) & (x <= 4)),
    (2*x/3 + 7/3, (x >= 4) & (x <= 7)))
```